Add tutorial for `sf::Angle` (#228)
SFML/SFML-Website: pages/tutorials/3.0/system/angle.md

This means that if the X-axis points to the right in the window and the Y-axis points down, then a positive angle rotates clockwise.
All SFML classes that work with angles use `sf::Angle`

All `sf::Angle` functions are `constexpr`-enabled so anything shown here can be done at compile-time.

## Constructing angles

There are two factory functions for constructing an angle: `sf::degrees` and `sf::radians`.

```cpp
sf::Angle angle1 = sf::degrees(180);
sf::Angle angle2 = sf::radians(3.1415f);
```

These two functions construct approximately equal angles from either degrees or radians so that you don't have to think about units in your own code.
Feel free to use either function as you see fit.
`sf::Angle` will handle any necessary unit conversions internally.

## Manipulating angles

`sf::Angle` provides a collection of operators for arithmetic operations.
You can add, subtract, multiply, and divide angles.
There are also modulo, equality, inequality, and comparison operators.

```cpp
sf::Angle angle1 = sf::degrees(10);
angle1 *= 2.f; // 20 degrees
sf::Angle angle2 = angle1 + sf::radians(0.5f); // 48.6 degrees
angle2 = -angle2; // -48.6 degrees

> angle2 /= 3.f; // 16.2 degrees

Reviewer: ```suggestion
angle2 /= 3.f; // -16.2 degrees
```

bool equal = (angle1 == angle2); // false
bool inequal = (angle1 != angle2); // true
```

`sf::Angle`s can exist outside range [-pi, pi) or [0, 2pi).
An angle may have a value of 720 degrees, for example.
If you would like to get a new angle wrapped to its equivalent value within a smaller range, there are two functions for doing so, `sf::Angle::wrapSigned` and `sf::Angle::wrapUnsigned`.

> - `wrapSigned` will return a new angle wrapped to the range [-pi, pi).

Reviewer: I think you need a blank line between the paragraph and the list for the list to be rendered as such.

Author: VS Code renders it fine but the renderer we use for the website doesn't behave identically so I'll an extra newline to be safe.

# Handling Angles

## Angles in SFML

> SFML defines angles as following the [right-hand rule](https://en.wikipedia.org/wiki/Right-hand_rule).
> This means that if the X-axis points to the right in the window and the Y-axis points down, then a positive angle rotates clockwise.
> Encoding these angles in code is handled by [`sf::Angle`](https://www.sfml-dev.org/todo/angle.php "sf::Angle documentation").
> All SFML classes that use angles use this class.
> `sf::Angle` represents an angle in a unit-agnostic manner so you can reason about angles in terms of degrees or radians.

Reviewer 2: The most important part is to explain what `sf::Angle` is, so I'd write it more like this:

```suggestion
[`sf::Angle`](https://www.sfml-dev.org/todo/angle.php "sf::Angle documentation") represents an angle in a unit-agnostic manner so you can reason about angles in terms of degrees or radians.
SFML defines angles by following the [right-hand rule](https://en.wikipedia.org/wiki/Right-hand_rule).
This means that if the X-axis points to the right in the window and the Y-axis points down, then a positive angle rotates clockwise.
All SFML classes that use angles work with `sf::Angle`.
```

All `sf::Angle` functions are `constexpr`-enabled so anything shown here can be done at compile-time.

## Constructing angles

> There are two ways to construct an angle: `sf::degrees` and `sf::radians`.

Reviewer 2: ```suggestion
There are two factory methods to construct an angle: `sf::degrees` and `sf::radians`.
```

```cpp

> sf::Angle angle1 = sf::degrees(15);
> sf::Angle angle2 = sf::radians(2);

Reviewer 2: ```suggestion
sf::Angle angle1 = sf::degrees(180);
sf::Angle angle2 = sf::radians(3.1415f);
```

```

These two functions construct the angle from either degrees or radians so that you don't have to think about units in your own code.
Feel free to use either function as you see fit.

> `sf::Angle` will handle all unit conversions internally.

Reviewer: This sounds like there were a lot of conversions, but there are none besides sf::degrees and sf::Angle::asDegrees.

## Manipulating angles

> `sf::Angle` provides a suite of operators for arithmetic operations.

Reviewer 2: ```suggestion
`sf::Angle` provides a collection of operators for arithmetic operations.
```

You can add, subtract, multiply, and divide angles.

> There are also modulo, (in)equalty, and comparison operators.

Reviewer 2: ```suggestion
There are also modulo, equality, inequality, and comparison operators.
```

```cpp
sf::Angle angle1 = sf::degrees(10);
angle1 *= 2.f; // 20 degrees

> angle1 + sf::radians(0.5f); // 48.6 degrees

Reviewer: This has no effect and might trigger a warning because [[nodiscard]]. Maybe store the result in a variable.

sf::Angle angle2 = -angle1; // -48.6 degrees
angle2 /= 3.f; // 16.2 degrees

bool equal = (angle1 == angle2); // false
bool inequal = (angle1 != angle2); // true
```

`sf::Angle`s can exist outside range [-pi, pi) or [0, 2pi).
An angle may have a value of 720 degrees, for example.

> If you would like to take an angle and wrap it to its equivalent value within a smaller range, there are two functions for doing so, `sf::Angle::wrapSigned` and `sf::Angle::wrapUnsigned`.

Reviewer: This paragraph is worded as if wrap methods were mutating in place, which is misleading.

> `wrapSigned` will take an angle and wrap it into the range [-pi, pi).
> `wrapUnsigned` will wrap an angle to the range [0, 2pi).

Reviewer 2: ```suggestion
- `wrapSigned` will take an angle and wrap it into the range [-pi, pi).
- `wrapUnsigned` will wrap an angle to the range [0, 2pi).
```

```cpp
sf::Angle angle1 = sf::degrees(540);
angle1.wrapUnsigned(); // 180 degrees

sf::Angle angle2 = sf::radians(2 * 3.1415f);
angle2.wrapSigned(); // 0 degrees
```

> After transformation the angles will occupy the same location on the unit circle.

Reviewer 2: ```suggestion
After transformation the angles will be at the same location on the unit circle.
```

Author: I'd prefer to keep "occupy" here if you don't mind

Reviewer 2: Fine by me 🙂

## User defined literals

C++ user-defined literals provide a convenient shorthand for writing angles.
These literals exist in the `sf::Literals` namespace.
Bring that namespace into the current scope to access them.

```cpp
using namespace sf::Literals;
sf::Angle angle1 = 45_deg; // 45 degrees
sf::Angle angle2 = angle1 + 3.1415_rad; // 225 degrees
```

## Accessing the underlying value

> If you would like to format an angle as text or pass the value to a function like `std::sin` there are two functions for accessing the angle as a raw `float`: `sf::Angle::asDegrees` and `sf::Angle::asRadians`.

Reviewer: Missing comma after std::sin
